app/uk/gov/hmrc/minorentityidentification/repositories/JourneyDataRepository.scala

```diff
@@ -77,10 +75,26 @@ class JourneyDataRepository @Inject()(mongoComponent: MongoComponent,
         Filters.equal(AuthInternalIdKey, authInternalId)
       ),
       Updates.unset(dataKey)
-    ).toFuture.map{
+    ).toFuture.map {
       _.getMatchedCount == 1
     }
 
+  def removeJourneyData(journeyId: String, authInternalId: String): Future[Boolean] =
+    collection.findOneAndReplace(
+      Filters.and(
+        Filters.equal(JourneyIdKey, journeyId),
+        Filters.equal(AuthInternalIdKey, authInternalId)
+      ),
+      Json.obj(
+        JourneyIdKey -> journeyId,
+        AuthInternalIdKey -> authInternalId,
+        CreationTimestampKey -> Json.obj("$date" -> Instant.now.toEpochMilli)
+      ),
+    ).toFutureOption.map {
+      case Some(_) => true
```

Comment:
... can we use here the same approach we used at line 79 ... or use the same approach for all ... using at line 79 and 67 the approach we use here...

Reply:
The methods on the collections return different things so we can't use the same method. findOneAndReplace returns the whole document that's been updated whereas the others return 'UpdateResult'

Reply:
Okay